_field-types/supported-field-types/derived.md

```diff
@@ -0,0 +1,903 @@
+          ],
+          "clientip": [
+            "61.177.2.0"
+          ],
+          "url": [
+            "/english/venues/images/venue_header.gif"
+          ]
+        },
+        "highlight": {
+          "url": [
+            "/english/venues/<em>images</em>/venue_header.gif"
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+## Performance
+
+Derived fields are not indexed but are computed dynamically by retrieving values from the `_source` field or doc values. Thus, they can be slow to execute. To improve performance, try the following:
+
+- Prune the search space by adding query filters on indexed fields together with derived fields.
+- Prefer using doc values over `_source` in the script for faster access, whenever applicable.
+- Consider using a [`prefilter_field`](#prefilter-field) to automatically prune the search space without explicit filters in the search request.
+
+### Prefilter field
+
+Specifying a prefilter field helps prune the search space automatically without adding explicit filters in the search request. The prefilter field specifies an existing indexed field (`prefilter_field`) on which to filter implicitly when constructing the query. The `prefilter_field` must be a text field (either [`text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/text/) or [`match_only_text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/match-only-text/)).
+
+For example, first update the mapping for the `method` derived field by specifying to prefilter on the `request` field: 
+
+```json
+PUT /logs/_mapping
+{
+  "derived": {
+    "method": {
+      "type": "keyword",
+      "script": {
+        "source": """
+        emit(doc["request.keyword"].value.splitOnToken(" ")[1])
+        """
+      },
+      "prefilter_field": "request"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Now search using a query on the `method` derived field:
+
+```json
+POST /logs/_search
+{
+  "profile": true,
+  "query": {
+    "term": {
+      "method": {
+        "value": "GET"
+      }
+    }
+  },
+  "fields": ["method"]
+}
+```
+{% include copy-curl.html %}
+
+OpenSearch implicitly adds a filter on the `request` field to your query:
+
+```json
+"#request:GET #DerivedFieldQuery (Query: [ method:GET])"
+```
+
+You can use the `profile` option can to analyze the derived field performance, as shown in the preceding example.
+{: .tip} 
+
+## Dervied object fields
+
+A script can emit a valid JSON object so you can query subfields without indexing them, in the same way as regular fields. This is useful for large JSON objects where occasional searches on some subfields are required. In this case, indexing the subfields is expensive while defining derived fields for each subfield also adds a lot of resource overhead. If you don't [explicitly provide the subfield type](#explicit-subfield-type), the subfield type is [inferred](#inferred-subfield-type).
```

Comment:
```suggestion
A script can emit a valid JSON object so that you can query subfields without indexing them, in the same way as with regular fields. This is useful for large JSON objects that require occasional searches on some subfields. In this case, indexing the subfields is expensive, while defining derived fields for each subfield also adds a lot of resource overhead. If you don't [explicitly provide the subfield type](#explicit-subfield-type), then the subfield type is [inferred](#inferred-subfield-type).
```